pywbem_mock/_inmemoryrepository.py

```diff
@@ -0,0 +1,431 @@
+        """
+        Overrides the abstract method.
+
+        Gets the object with name from the dictionary or returns an exception
+        if an object with name does note exist.
+
+        If copy is True, it generates creates a copy and returns the copy
+        """
+
+        # pylint: disable=no-else-return
+        if name in self._data:
+            if copy:
+                return self._data[name].copy()
+            return self._data[name]
+        else:
+            raise KeyError('{} not in {} object store'
+                           .format(name, self._cim_object_type))
+
+    def create(self, name, cim_object):
+        """
+        Overrides the abstract method.
+
+        Creates a new cim_object in the object store dictionary or fails
+        with an exception if the name already exists in the dictionary.
+        """
+
+        assert isinstance(cim_object, self._cim_object_type)
+
+        if name in self._data:
+            raise ValueError('{} already in {} object store'
+                             .format(name, self._cim_object_type))
+
+        self._data[name] = cim_object.copy()
+
+    def update(self, name, cim_object):
+        """
+        Overrides the abstract method.
+
+        This method updates the corresponding dictionary with the cim_object
+        if the object exists in the dictionary. Otherwise it returns
+        an exception.
+
+        """
+
+        assert isinstance(cim_object, self._cim_object_type)
+
+        if name not in self._data:
+            raise KeyError('{} not in {} object store'
+                           .format(name, self._cim_object_type))
+
+        # Replace the existing object with a copy of the input object
+        self._data[name] = cim_object.copy()
+
+    def delete(self, name):
+        """
+        Overrides the abstract method.
+
+        If the object exists in the object store dictionary, it is deleted.
+        If it does not exist an exception is returned.
+        """
+        if name in self._data:
+            del self._data[name]
+        else:
+            raise KeyError('{} not in {} object store'
+                           .format(name, self._cim_object_type))
+
+    def iter_names(self):
+        """
+        Overrides the abstract method.
+
+        Returns iterator for names of the objects in the object store
+
+        """
+        return six.iterkeys(self._data)
+
+    def iter_values(self, copy=True):
+        """
+        Overrides the abstract method.
+
+        Returns an iterator for the object values in the object store
+        """
+        for value in six.itervalues(self._data):
+            if copy:
+                yield(value.copy())
+            else:
+                yield(value)
+
+    def len(self):
+        """
+        Overrides the abstract method.
+
+        :class:`~pywbem_mock._baserepository.ObjectStoreAPI.len`
+
+        Returns count of number of objects in the object store dictionary.
+        """
```

Comment:
Add sections Parameters, Returns and Raises.

Reply:
See comment above